src/compat/dbtype_function.h

```diff
@@ -0,0 +1,501 @@
+#define DB_GET_STRING_LENGTH(v) db_get_string_length(v)
+#define DB_GET_DOUBLE(v) db_get_double(v)
+#define DB_GET_OBJECT(v) db_get_object(v)
+#define DB_GET_SET(v) db_get_set(v)
+#define DB_GET_MIDXKEY(v) db_get_midxkey(v)
+#define DB_GET_POINTER(v) db_get_pointer(v)
+#define DB_GET_TIME(v) db_get_time(v)
+#define DB_GET_TIMETZ(v) db_get_timetz(v)
+#define DB_GET_TIMESTAMP(v) db_get_timestamp(v)
+#define DB_GET_TIMESTAMPTZ(v) db_get_timestamptz(v)
+#define DB_GET_DATETIME(v) db_get_datetime(v)
+#define DB_GET_DATETIMETZ(v) db_get_datetimetz(v)
+#define DB_GET_DATE(v) db_get_date(v)
+#define DB_GET_MONETARY(v) db_get_monetary(v)
+#define DB_GET_ERROR(v) db_get_error(v)
+#define DB_GET_ELO(v) db_get_elo(v)
+#define DB_GET_NUMERIC(v) db_get_numeric(v)
+#define DB_GET_BIT(v, l) db_get_bit(v, l)
+#define DB_GET_CHAR(v, l) db_get_char(v, l)
+#define DB_GET_NCHAR(v, l) db_get_nchar(v, l)
+#define DB_GET_STRING_SIZE(v) db_get_string_size(v)
+#define DB_GET_ENUM_SHORT(v) db_get_enum_short(v)
+#define DB_GET_ENUM_STRING(v) db_get_enum_string(v)
+#define DB_GET_ENUM_STRING_SIZE(v) db_get_enum_string_size(v)
+#define DB_GET_METHOD_ERROR_MSG() db_get_method_error_msg()
+#define DB_GET_RESULTSET(v) db_get_resultset(v)
+#define DB_GET_STRING_CODESET(v) ((INTL_CODESET) db_get_string_codeset(v))
+#define DB_GET_STRING_COLLATION(v) db_get_string_collation(v)
+#define DB_GET_ENUM_CODESET(v) db_get_enum_codeset(v)
+#define DB_GET_ENUM_COLLATION(v) db_get_enum_collation(v)
+#define DB_VALUE_TYPE(value) db_value_type(value)
+#define DB_VALUE_PRECISION(value) db_value_precision(value)
+#define DB_VALUE_SCALE(value) db_value_scale(value)
+
+#define DB_GET_INTEGER(value)           db_get_int(value)
+#define DB_GET_BIGINTEGER               DB_GET_BIGINT
+#define DB_GET_OBJ DB_GET_OBJECT
+#define DB_GET_MULTISET(value)          db_get_set(value)
+#define DB_GET_LIST(value)              db_get_set(value)
+#define DB_GET_SEQUENCE DB_GET_LIST
+#define DB_GET_COLLECTION(value)        db_get_set(value)
+#define DB_GET_UTIME DB_GET_TIMESTAMP
+#define DB_GET_SMALLINT(value)          db_get_short(value)
+
+#define DB_GET_COMPRESSED_SIZE(value) db_get_compressed_size(value)
+
+#define DB_GET_JSON_DOCUMENT(value) db_get_json_document(value)
+
+#define DB_GET_SEQ DB_GET_SEQUENCE
+
+#define DB_SET_COMPRESSED_STRING(value, compressed_string, compressed_size, compressed_need_clear) \
+	db_set_compressed_string(value, compressed_string, compressed_size, compressed_need_clear)
+
+#define DB_TRIED_COMPRESSION(value) (DB_GET_COMPRESSED_SIZE(value) != DB_NOT_YET_COMPRESSED)
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+  extern DB_TYPE setobj_type (COL * set);
+  /********************************************************/
+  /* From elo.h */
+
+  extern void elo_init_structure (DB_ELO * elo);
+  /********************************************************/
+  /* From db_date.h */
+
+  extern int db_date_encode (DB_DATE * date, int month, int day, int year);
```

Comment:
Duplicate declaration of db_time_encode in dbi_compat.h. Please move all date/time encode/decode declarations here.